pdp/handlers_upload.go

```diff
@@ -0,0 +1,285 @@
+package pdp
+
+import (
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path"
+
+	"github.com/go-chi/chi/v5"
+	"github.com/google/uuid"
+	logger "github.com/ipfs/go-log/v2"
+	"github.com/yugabyte/pgx/v5"
+
+	commcid "github.com/filecoin-project/go-fil-commcid"
+	commp "github.com/filecoin-project/go-fil-commp-hashhash"
+	"github.com/filecoin-project/go-state-types/abi"
+
+	"github.com/filecoin-project/curio/harmony/harmonydb"
+	"github.com/filecoin-project/curio/lib/dealdata"
+)
+
+var log = logger.Logger("pdp")
+
+// PieceSizeLimit in bytes
+var PieceSizeLimit = abi.PaddedPieceSize(256 << 20).Unpadded()
+
+func (p *PDPService) handlePiecePost(w http.ResponseWriter, r *http.Request) {
+	// Verify that the request is authorized using ECDSA JWT
+	serviceID, err := p.verifyJWTToken(r)
+	if err != nil {
+		http.Error(w, "Unauthorized: "+err.Error(), http.StatusUnauthorized)
+		return
+	}
+
+	// Parse request body
+	var req struct {
+		PieceCID string `json:"pieceCid"`
```

Comment:
It looks like you're not requiring piececid v2 here, not in the put handler. Long term, I think we should require this. From the standpoint of the SP, I imagine them to want to filter at the piece level, and I imagine one of the main filter critieria to be size. They should know exactly what size piece they're about to accept before they accept it. 

Unless I'm wrong, a PieceCID v1 doesn't specify that, and absent a size parameter, they could be accepting anything from 32 bytes to 256 MB.

Reply:
Yes, PieceCIDv2 is still todo, can quickly update if that's a blocker